js/test/generate-test-data.ts

```diff
@@ -512,30 +518,43 @@ function generateStruct<T extends Struct>(this: TestDataVectorGenerator, type: T
     const values = memoize(() => {
         const values = [] as any[];
         const childValues = cols.map((x) => x());
+        const names = type.children.map((f) => f.name);
         iterateBitmap(length, nullBitmap, (i, valid) => {
-            values[i] = !valid ? null : childValues.map((col) => col[i]);
+            values[i] = !valid ? null : childValues.reduce((row, col, j) => ({
+                ...row, [names[j]]: col[i]
+            }), {});
         });
         return values;
     });
     return { values, vector: Vector.new(Data.Struct(type, 0, length, nullCount, nullBitmap, vecs)) };
 }
 
-function generateMap<T extends Map_>(this: TestDataVectorGenerator, type: T, length = 100, nullCount = length * 0.2 | 0, children = type.children.map((f) => this.visit(f.type, length, nullCount))): GeneratedVector<V<T>> {
-    const vecs = children.map(({ vector }) => vector);
-    const cols = children.map(({ values }) => values);
+function generateMap<T extends Map_>(this: TestDataVectorGenerator,
```

Comment:
What do you think about adding a simple test for MapVector to vector-tests.ts as well? The generated data tests are nice for completeness, but I like that vector-tests serve as a form of documentation and a sanity check in case anything goes wrong in this test